### DISCUSSION TITLE
Has anyone worked with the package ulmo?

### DISCUSSION THREAD

**Discussion by millie-spencer** [earthlab-education/Earth-Analytics-2023-01-Intro#508] [Q&A06 Vector Data]
I'm working on my repeat analysis of the wildfire data from this week, and am hoping to play with SNOTEL data so that I can assess how snow variables (e.g. snow-water equivalent) are related to wildfire frequency and magnitude. I found a great tutorial from the SnowEx hackweek (https://snowex-2021.hackweek.io/tutorials/geospatial/SNOTEL_query.html), but the analysis requires a package called 'ulmo' which I can't seem to install. I tried installing the package and got the following error. Lmk if you have any ideas! 
<img width="1202" alt="Screenshot 2023-10-18 at 12 17 01 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/127904183/452bbf29-8666-4664-93e6-90f2b4ee440f">

**Comment by eculler** [comment]
Hi @millie-spencer , I have not worked with `ulmo`. However, I have a few suggestions.

First off, I would try installing this package with conda (or mamba, for faster results):

```bash
conda install -c conda-forge ulmo
```

To use `mamba`, install mamba and then replace `conda` with `mamba`.

`conda-forge` tends to be more stable than `pip`.

I also notice that one of the dependencies may have been written in Python2 (it is running use_2to3)...this is unfortunately not a good sign for getting it to install, especially if you have an M1 Mac, although `ulmo` itself seems to be very actively maintained.

Finally, the folks at CUAHSI are very friendly and know a lot about getting environments to work from their Hydroshare project. It's worth reaching out if the `conda` install doesn't work for you.

**Reply by millie-spencer** [reply]
Hi Elsa! Unfortunately I already tried to conda-forge approach, it looks like some dependency or default setting with our codespace is interfering with the ulmo package. I might try setting up a new codespace and see if it works!
